### ISSUE TITLE
[Enhancement] Don't spam the connect tab and notification if user dismisses it more than once

### ISSUE THREAD

**Issue by levicki** [kee-org/browser-addon#316]
Sometimes Kee will ask to connect to Keepass database again by requesting connection password (it happened to me just now in Brave even though I was using it for a long time with no issues).

If I close the tab with the password prompt it will reopen itself in a few seconds, and it will do that incessantly again and again until I relent and do what it wants. Not only that, but it will also spam some toast notifications every time the tab is auto-opened again, even though I have disabled notifications both in the browser for all websites and in Windows for all applications.

I can't even begin to describe how annoying and disrespectful that behavior is -- it borders on a dark pattern and reminds me of malware.

TL;DR -- you can't just hijack the whole browser and nag until whatever you want users to do with your extension is complete, because the users might have more pressing things to do in that browser at the moment.

I have uninstalled Kee until it stops behaving like my computer is yours.

**Comment by luckyrat** [comment]
Unless something major has changed with the way operating systems, browsers and KeePass work in recent years, I do not think there is any alternative to the current approach and there is little to no drawback to either re-authorising the connection when required or disabling Kee until you next want to use it.

We'll look into alternatives using WebExtensions native messaging soon but I don't know whether that can work with a KeePass plugin, nor whether it will help with this specific behaviour, but it might.

This might seem like a trivial case of a developer not considering what's best for users but it ultimately is a highly complex topic with many factors to consider across at least 5 different pieces of software, much of which is not under our control. Feel free to walk us through any alternative ideas you have in this area and/or get involved with any discussions about native messaging when they happen on the community forum.

By the way, in future, using the [community forum](https://forum.kee.pm/) will likely result in faster help (I've just updated to the latest GitHub issue template system to try to make that clearer before one takes the time to write up an issue here).
